addons/external_intercom/functions/fnc_connect.sqf

```diff
@@ -0,0 +1,148 @@
+  Arguments:
+    0: Vehicle object <OBJECT>
+    1: Player unit <OBJECT>
+    2: Is Wireless connection? <BOOLEAN>
+
+  Return Value:
+    None
+
+  Example:
+    [_vehicle, _player] call TFAR_external_intercom_fnc_connect;
+
+  Public: Yes
+*/
+
+params ["_vehicle", "_player", "_actionParams"];
+_actionParams params [["_wireless", false, [false]]];
+
+if (_wireless && !((headgear _player) in TFAR_externalIntercomWirelessHeadgear)) exitWith {
+    [parseText format[CSTRING(HEADGEAR_NOT_WIRELESS_CAPABLE), getText(configFile >> "CfgWeapons" >> headgear _player >> "displayName")]] call TFAR_fnc_showHint;
+    "no";
+};
+
+private _externalSpeakers = _vehicle getVariable ["TFAR_ExternalIntercomSpeakers", [objNull, []]];
+// External Speakers structure is always:
+//  [_playerUsingPhone, [_infinitePlayersConnectedWirelessly...]]
+if (!isNull (_externalSpeakers select 0) && !_wireless) exitWith {
+    [parseText format[CSTRING(SOMEONE_USING_PHONE), name (_externalSpeakers select 0)]] call TFAR_fnc_showHint;
+    "no";
+};
+
+// Make sure we disconnect properly from any other connected vehicles
+private _oldVehicle = _player getVariable "TFAR_ExternalIntercomVehicle";
+if (!isNil {_oldVehicle}) then {
+    [_oldVehicle, _player] call TFAR_external_intercom_fnc_disconnect;
+};
+
+private _vehicleID = _vehicle getVariable ["TFAR_vehicleIDOverride", netId _vehicle];
+_vehicle setVariable ["TFAR_vehicleIDOverride", _vehicleID, true];
+_player setVariable ["TFAR_ExternalIntercomVehicle", _vehicle, true];
+_player setVariable ["TFAR_vehicleIDOverride", _vehicleID, true];
+
+if (_wireless) then {
+    (_externalSpeakers select 1) pushBack _player;
+} else {
+    _externalSpeakers set [0, _player];
+};
+_vehicle setVariable ["TFAR_ExternalIntercomSpeakers", _externalSpeakers, true];
+
+// Get intercom channel, now that we've signed on to the intercom
+private _intercomSlot = [_vehicle] call TFAR_external_intercom_fnc_getIntercomChannel;
+
+_vehicle setVariable [format ["TFAR_IntercomSlot_%1",(netId _player)], _intercomSlot, true];
+_player setVariable [format ["TFAR_IntercomSlot_%1",(netId _player)], _intercomSlot, true]; // Because TFAR checks the "vehicle", and thinks the unit is the vehicle when it's not in a vehicle, so we manually set it
+
+// Show indicator
+if (TFAR_oldVolumeHint) then {
+    [parseText "Connected to Vehicle Intercom", -1] call TFAR_fnc_showHint;
+} else {
+    (QGVAR(PhoneConnectionIndicatorRsc) call BIS_fnc_rscLayer) cutRsc [QGVAR(PhoneConnectionIndicatorRsc), "PLAIN", 0, true];
+};
+
+// Keep connection alive until out of range or disconnected
+private _interactionPointRelative = getCenterOfMass _vehicle;
+private _intercomMaxRange = TFAR_externalIntercomMaxRange_Wireless;
+private _headgear = "";
+
+if (!_wireless) then {
+    _interactionPointRelative = [_vehicle] call TFAR_external_intercom_fnc_getInteractionPoint;
+    _intercomMaxRange = TFAR_externalIntercomMaxRange_Phone;
+} else {
+    _headgear = headgear _player;
+};
+
+[{
```

Comment:
I'd suggest optimizing this by writing two conditions, one for wireless and one for not wireless.

You don't need modelToWorld (somewhat expensive) if its wireless, as center of mass is usually 0,0,0 so you can just use vehicle pos.

You also don't need to do the wireless check if not wireless.

You can do one check here, to save performance on every frame